File: tri_js/exo-tri.js
var tableau = [90, 88, 50, 45, 78, 147, 37, 56, 4];
var echange = 0;
var trier = false;
var i;

while (trier === false) {
	trier = true;
	for (i = 0; i < tableau.length /2; i++) {
		if (tableau[i] > tableau[i + 1]) {
			echange = tableau[i];
			tableau[i] = tableau[i + 1];
			tableau[i + 1] = echange;
			trier = false;
		}
	}
}
document.write(tableau);

File: tri_ruby/exo-tri.rb
tableau = [90, 88, 50, 45, 78, 147, 37, 56, 4]
echange = 0
trier = 0

while trier == 0  do
    trier = 1
    i=0    
    while i<tableau.length-1
        if tableau[i] > tableau[i+1]
            echange=tableau[i]
            tableau[i]=tableau[i+1]
            tableau[i+1]=echange
            trier=0
        end
        i=i+1
    end
    
end

puts tableau